test/tests.zig

```diff
@@ -769,6 +769,54 @@ pub fn addCliTests(b: *std.Build) *Step {
         step.dependOn(&cleanup.step);
     }
 
+    {
```

Comment:
This set of tests belongs in linker tests in `test/link/elf.zig`. I also don't understand why can't it be orchestrated using the build system? Why do you need to invoke zig exe via `addSystemCommand`? `exe.linkSystemLibrary("foo")` emits as `-lfoo` fwiw. Did you actually have a look at `test/link/elf.zig`? We do have a substantial amount of tests that rely on emitting and linking against system libs.

Reply:
I've moved the testing to `test/link/elf.zig`. The new commits add a few bits of extra functionality that are needed to reproduce the problem.

Reply:
@kubkon I think I've addressed your comments on this PR, any thoughts?

Reply:
@andrewrk @kubkon Could you please have a look at this PR?